package/src/utils/DBSyncManager.ts

```diff
@@ -62,7 +75,15 @@ export class DBSyncManager {
       this.listeners.forEach((l) => l(true));
     }
 
-    this.client.on('connection.changed', async (event) => {
+    // If a listener has already been registered, unsubscribe from it so
+    // that it can be reinstated. This can happen if we reconnect with a
+    // different user for example, which would otherwise cause the old
+    // listener to produce a memory leak.
+    if (this.connectionChangedListener) {
+      this.connectionChangedListener.unsubscribe();
+    }
```

Comment:
Could we also somehow make sure that unsubscribe is called when the component is unmounted

Reply:
For sure, initially I was going to only do that but then figured this should probably be done within the `DBSyncManager` itself (i.e be in control of its own listener state when initializing). But I will add it just for safety, thanks !